styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -171,7 +193,9 @@ private void ensureSecret(WorkflowInstance workflowInstance, RunSpec runSpec) {
                                     .build());
         LOG.info("[AUDIT] Secret {} created for service account {}", secretName, serviceAccount);
       }
-    } else if (runSpec.secret().isPresent()) {
+    }
+
+    if (runSpec.secret().isPresent()) {
```

Comment:
I was thinking that we'd let them coexist. We want to gradually migrate and the old-style secret still makes sense for oss?